api/v1alpha1/inferencepool_types.go

```diff
@@ -59,8 +59,85 @@ type InferencePoolSpec struct {
 	// +kubebuilder:validation:Maximum=65535
 	// +kubebuilder:validation:Required
 	TargetPortNumber int32 `json:"targetPortNumber"`
+
+	// EndpointPickerConfig configures the extension that runs the endpoint picking service.
+	// to this pool.
+	EndpointPickerConfig `json:"endpointPickerConfig"`
+}
+
+type EndpointPickerConfig struct {
+	// Extension configures an endpoint picker as an extension service.
+	//
+	// +optional
+	Extension *ExtensionConfig `json:"extension"`
+}
+
+// ExtensionConfig specifies how to configure an extension that runs the endpoint picker.
+type ExtensionConfig struct {
+	// ExtensionRef is a reference to a service extension.
+	ExtensionRef *ExtensionReference `json:"extensionRef"`
+
+	// ExtensionConnection configures the connection between the gateway and the extension.
+	ExtensionConnection `json:"extensionConnection"`
+}
+
+// ExtensionReference is a reference to the extension deployment.
+type ExtensionReference struct {
+	// Group is the group of the referent.
+	// When unspecified or empty string, core API group is inferred.
+	//
+	// +optional
+	// +kubebuilder:default=""
+	Group *string `json:"group,omitempty"`
+
+	// Kind is the Kubernetes resource kind of the referent. For example
+	// "Service".
+	//
+	// Defaults to "Service" when not specified.
+	//
+	// ExternalName services can refer to CNAME DNS records that may live
+	// outside of the cluster and as such are difficult to reason about in
+	// terms of conformance. They also may not be safe to forward to (see
+	// CVE-2021-25740 for more information). Implementations MUST NOT
+	// support ExternalName Services.
+	//
+	// Support: Core (Services with a type other than ExternalName)
+	//
+	// Support: Implementation-specific (Services with type ExternalName)
+	//
+	// +optional
+	// +kubebuilder:default=Service
+	Kind *string `json:"kind,omitempty"`
+
+	// Name is the name of the referent.
+	Name string `json:"name"`
+
+	// The port number on the pods running the extension. When unspecified, implementations are recommended
+	// to default it to 9002 and the Kind is Service.
+	TargetPortNumber *int32 `json:"targetPortNumber"`
 }
 
+// ExtensionConnection encapsulates options that configures the connection to the extension.
+type ExtensionConnection struct {
+	// Configures how the gateway handles the case when the extension is not responsive.
+	// Defaults to failClose.
+	//
+	// +kubebuilder:default="FailClose"
+	FailureMode ExtensionFailureMode `json:"failureMode"`
```

Comment:
It doesn't make sense to use `+kubebuilder:default` on a non-pointer field. The field will always have a value and cannot be left unset.

Reply:
To clarify, I think you're arguing for this change?
```suggestion
	// +optional
	// +kubebuilder:default="FailClose"
	FailureMode *ExtensionFailureMode `json:"failureMode,omitempty"`
```